src/librados/IoCtxImpl.cc

```diff
@@ -747,6 +747,65 @@ int librados::IoCtxImpl::operate_read(const object_t& oid,
   return r;
 }
 
+int librados::IoCtxImpl::aio_operate_repair_read(const object_t& oid,
+				      ::ObjectOperation *o,
+				      AioCompletionImpl *c,
+				      bufferlist *pbl,
+				      int flags, int32_t osdid, epoch_t e, int op_flags)
+{
+  if (!o->size())
+    return 0;
+
+  Context *onack = new C_aio_Ack(c);
+
+  c->is_read = true;
+  c->io = this;
+
+  version_t ver;
+
+  int op = o->ops[0].op.op;
+  ldout(client->cct, 10) << ceph_osd_op_name(op) << " oid=" << oid << " nspace=" << oloc.nspace << dendl;
+  // Prepend assert_interval op
+  OSDOp tmp;
+  tmp.op.op = CEPH_OSD_OP_ASSERT_INTERVAL;
+  tmp.op.assert_interval.epoch = e;
+  o->ops.insert(o->ops.begin(), tmp);
+  int size = o->ops.size();
+  o->out_bl.resize(size);
+  o->out_handler.resize(size);
+  o->out_rval.resize(size);
+  o->out_bl[1] = pbl;
+  o->ops[1].op.flags = op_flags;
+  Objecter::Op *objecter_op = objecter->prepare_read_op(oid, oloc,
+	                                      *o, snap_seq, NULL,
+	                                      flags | CEPH_OSD_FLAG_REPAIR_READS | CEPH_OSD_FLAG_IGNORE_OVERLAY,
+	                                      onack, &ver);
+  objecter_op->target.osd = osdid;
```

Comment:
minor, 
we don't guard the case where osd is not up in `Objecter::op_submit()`. maybe we can check it in `Objecter::_calc_target()`? i have following backtrace if an invalid osdid is passed in:

```
 ceph version Development (no_version)
 1: (ceph::BackTrace::BackTrace(int)+0x2d) [0x4c5e6d]
 2: ./bin/rados() [0x4c5540]
 3: (()+0x10d30) [0x7f0fcae36d30]
 4: (gsignal()+0x38) [0x7f0fbba9b458]
 5: (abort()+0x16a) [0x7f0fbba9c8da]
 6: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x310) [0x7f0fc1156e8b]
 7: (OSDMap::get_inst(int) const+0x48) [0x7f0fc0a1fd2c]
 8: (Objecter::_get_session(int, Objecter::OSDSession**, ceph::shunique_lock<boost::shared_mutex>&)+0x4fc) [0x7f0fc09f9f1c]
 9: (Objecter::_op_submit(Objecter::Op*, ceph::shunique_lock<boost::shared_mutex>&, unsigned long*)+0x2a0) [0x7f0fc09ff290]
 10: (Objecter::_op_submit_with_budget(Objecter::Op*, ceph::shunique_lock<boost::shared_mutex>&, unsigned long*, int*)+0x266) [0x7f0fc09fe7c4]
 11: (Objecter::op_submit(Objecter::Op*, unsigned long*, int*)+0x82) [0x7f0fc09fe4ec]
 12: (librados::IoCtxImpl::aio_operate_repair_read(object_t const&, ObjectOperation*, librados::AioCompletionImpl*, ceph::buffer::list*, int, int, unsigned int, int)+0x458) [0x7f0f
c09cfddc]
 13: (librados::IoCtxImpl::operate_repair_read(object_t const&, ObjectOperation*, ceph::buffer::list*, int, int, unsigned int, int)+0x6c) [0x7f0fc09cfed8]
 14: (librados::IoCtxImpl::repair_read(object_t const&, ceph::buffer::list&, unsigned long, unsigned long, int, int, unsigned int)+0xf4) [0x7f0fc09d2a44]
 15: (librados::IoCtx::repair_read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ceph::buffer::list&, unsigned long, unsigned long, int, i
nt, unsigned int)+0x6e) [0x7f0fc0963738]
 16: ./bin/rados() [0x475100]
 17: ./bin/rados() [0x47f22f]
 18: (main()+0x2294) [0x48c867]
 19: (__libc_start_main()+0xf0) [0x7f0fbba885f0]
 20: (_start()+0x29) [0x4745b9]
 NOTE: a copy of the executable, or `objdump -rdS <executable>` is needed to interpret this.
```